src/library-authoring/library-access/LibraryAccessForm.jsx

```diff
@@ -0,0 +1,180 @@
+    <Row className="mb-2">
+      <form className="col-12" onSubmit={onSubmit}>
+        <Card>
+          <Card.Body>
+            <div className="form-create">
+              <Card.Title>Grant Access to This Library</Card.Title>
+              <fieldset>
+                <ol className="list-input">
+                  <li className="field">
+                    <ValidationFormGroup
+                      for="title"
+                      helpText={intl.formatMessage(messages['library.access.form.email.help'])}
+                      invalid={hasFieldError('email')}
+                      invalidMessage={getFieldError('email')}
+                      className="mb-0 mr-2"
+                    >
+                      <label className="h6 d-block" htmlFor="email">
+                        {intl.formatMessage(messages['library.access.form.email.label'])}
+                      </label>
+                      <Input
+                        name="email"
+                        id="email"
+                        type="text"
+                        placeholder={intl.formatMessage(messages['library.access.form.email.placeholder'])}
+                        value={data.email}
+                        onChange={onValueChange}
+                      />
+                    </ValidationFormGroup>
+                  </li>
+                </ol>
+              </fieldset>
+            </div>
+          </Card.Body>
+        </Card>
+        <Card className="mb-5">
+          <Card.Body>
+            <StatefulButton
+              variant="primary"
+              type="submit"
+              size="lg"
+              state={getSubmitButtonState()}
+              labels={{
+                disabled: intl.formatMessage(messages['library.access.form.button.submit']),
+                enabled: intl.formatMessage(messages['library.access.form.button.submit']),
+                pending: intl.formatMessage(messages['library.access.form.button.submitting']),
+              }}
+              icons={{
+                pending: <Icon className="fa fa-spinner fa-spin" />,
+              }}
+              disabledStates={['disabled', 'pending']}
+              className="action-button"
+            />
+            <Button size="lg" variant="secondary" className="mx-3 action-button" onClick={() => setShowAdd(false)}>Cancel</Button>
+          </Card.Body>
+        </Card>
+      </form>
+    </Row>
+  </>
+);
+
+LibraryAccessForm.propTypes = {
+  intl: intlShape.isRequired,
+  // eslint-disable-next-line react/forbid-prop-types
+  data: PropTypes.object.isRequired,
+  onValueChange: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  setShowAdd: PropTypes.func.isRequired,
+  hasFieldError: PropTypes.func.isRequired,
+  getFieldError: PropTypes.func.isRequired,
+  getSubmitButtonState: PropTypes.func.isRequired,
+};
+
+const initialFormState = () => ({
+  email: '',
+  access_level: LIBRARY_ACCESS.USER,
+});
+
+const LibraryAccessFormContainer = (
```

Comment:
Nit/optional: I recommend putting a JSDoc-style comment above every React component, with a couple lines explaining what it does and what other components it interacts with.

e.g. 
```
/**
 * LibraryAccessFormContainer:
 * Widget for editing the list of users with access to this content library.
 * This wraps LibraryAccessForm and handles API calls to fetch/write data, as well as displaying 
 * a loading message during initial loading. This separation allows LibraryAccessForm to be tested
 * in unit tests without needing to mock the API.
 */